test/Amadevus.RecordGenerator.Test/RecordMutatorTests.cs

```diff
@@ -27,6 +28,13 @@ public void With_SimpleProperty_CreatesModifiedInstance()
             Assert.Equal(newName, modifiedItem.Name);
         }
 
+        [Fact]
+        public void Ctor_InvokesValidate_Passing()
+        {
+            var item = CreateItem();
+            Assert.Throws<ArgumentNullException>("value", () => item.WithName(null));
```

Comment:
I'm not sure what you're trying to demo. Argument name passed to `ArgumentNullException` should reflect argument name at the place of check being done. The check is done in constructor via delegation to partial `Validate` method.

`Validate` body is written by the user, doing whatever he'll want to. And in this case I'd say that calling out Property name for the exception is as correct as it gets, because the ctor right now is just a bridge between caller and property setter. Ideally, this'd be done in the object initializer, but we can't use it 'cause we're immutable.

So to sum up, let's use nameof(Property) for our tests. In user guidelines that's also what we can advertise. And then, in the end, it's up to a developer to write the exception code.

Reply:
See 4076ebde486b317355b0a1e2b2e9f935a53a4050 for a compromise.

Reply:
I'll stay with the `nameof(Property)`. It's more maintainable for our tests, and doesn't strike me as misleading anyway.

Reply:
Done with 1ff6ba45e45245343dcc73510986db2dd5f976ba.